### PR TITLE
Implement library-clean-up

### PR DESCRIPTION
**Pull request by ColdenCullen**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/mqtt/mqtt.h [hunk 1]
```diff
@@ -88,6 +88,9 @@ void aws_mqtt_library_init(struct aws_allocator *allocator);
 AWS_MQTT_API
 void aws_mqtt_library_clean_up(void);
 
+AWS_MQTT_API
+void aws_mqtt_fatal_assert_library_initialized(void);
```
Comment (justinboswell) [inline include/aws/mqtt/mqtt.h]: Are we going to call this anywhere?

Comment (ColdenCullen) [reply include/aws/mqtt/mqtt.h]: 🤷‍♂ Nothing has a hard dependency on it yet, but @graebm wanted it present

Comment (justinboswell) [reply include/aws/mqtt/mqtt.h]: Maybe just shove it in client create and call it good enough for now?

Comment (ColdenCullen) [reply include/aws/mqtt/mqtt.h]: Sure #yolo

**Pull request merged**